src/notebooks/controllers/notebookControllerManager.ts

```diff
@@ -324,9 +357,19 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         // we don't end up with a kernel that asks to install python)
         if (!this.isWeb) {
             if ([...this.registeredControllers.values()].some((item) => isPythonKernelConnection(item.connection))) {
-                this.removeNoPythonControllers();
+                // We have some type of python kernel, turn off both install helper commands
+                await this.showInstallPythonExtensionContext.set(false);
+                await this.showInstallPythonContext.set(false);
             } else {
-                this.registerNoPythonControllers();
+                if (!this.extensionChecker.isPythonExtensionInstalled) {
```

Comment:
I think this too can be moved into a serpate class.
Today we do expose the controllers via a public property and also have a method to load all of the controllers & wait for it to complete loading.

Basically, I'd like to ensure this complex class doesn't get more complicated (with additional logic & processing)